### PR TITLE
STORM-2665: Adapt Kafka's release note generation script for Storm

### PR DESCRIPTION
**Pull request by srdo**
See https://issues.apache.org/jira/browse/STORM-2665

### CODE REVIEW

**Comment by hmcl** [comment]
@srdo Can you please add a screenshot, or a link for a sample of the output. Thanks.

**Comment by srdo** [comment]
Whoops, seems like there's a different jira module in the directory I put the script in than I expected. Will move it. Here's some sample output after running `$ ./release_notes.py 1.1.1 > relnotes.html` https://paste.apache.org/JFDP

**Comment by HeartSaVioR** [comment]
+1 for the change, and it would be better to rename existing python package directories and include this to dev-tools directory anyway.

**Comment by srdo** [comment]
I think this is good to go. @ptgoetz would this script work for you?

**Comment by HeartSaVioR** [comment]
+1 again, and waiting for @ptgoetz opinion.

File: dev-tools/release_notes.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    # type a bit to ensure features and improvements end up first.
+    def issue_type_key(issue):
+        if issue.fields.issuetype.name == 'New Feature':
+            return -2
+        if issue.fields.issuetype.name == 'Improvement':
+            return -1
+        return issue.fields.issuetype.id
+    by_group = [(k,sorted(g, key=lambda issue: issue.id)) for k,g in itertools.groupby(sorted(issues, key=issue_type_key), lambda issue: issue.fields.issuetype.name)]
+
+    print "<h1>Release Notes - Storm - Version %s</h1>" % version
```
Comment (hmcl) [inline dev-tools/release_notes.py]: @srdo I would put the title as **Storm 1.1.1 Release Notes** or **Release Notes for Storm 1.1.1**

File: dev-tools/release_notes.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    print """<p>Below is a summary of the JIRA issues addressed in the %(version)s release of Storm. For full documentation of the
```
Comment (hmcl) [inline dev-tools/release_notes.py]: JIRA issues addressed in the Storm %(version)s release.  Documentation for this release is available in the Apache Storm project site.

File: dev-tools/release_notes.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+"""Usage: release_notes.py <version> > RELEASE_NOTES.html
+
+Generates release notes for a Storm release by generating an HTML doc containing some introductory information about the
+ release with links to the Storm docs followed by a list of issues resolved in the release. The script will fail if it finds
+ any unresolved issues still marked with the target release. You should run this script after either resolving all issues or
+ moving outstanding issues to a later release.
+
+"""
+
+from jira import JIRA
```
Comment (ptgoetz) [inline dev-tools/release_notes.py]: Shouldn't this be `from jira_custom import JiraRepo`?

File: dev-tools/release_notes.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        results += new_results
+        startAt += len(new_results)
+    return results
+
+def issue_link(issue):
+    return "%s/browse/%s" % (JIRA_BASE_URL, issue.key)
+
+
+if __name__ == "__main__":
+    apache = JIRA(JIRA_BASE_URL)
```
Comment (ptgoetz) [inline dev-tools/release_notes.py]: `apache = JiraRepo(JIRA_BASE_URL)` ?

File: dev-tools/release_notes.py (hunk 3 above)
Comment (srdo) [inline dev-tools/release_notes.py]: No, the script uses https://pypi.python.org/pypi/jira/. I'll add a comment to the top that users should install that module.

**Comment by srdo** [comment]
Addressed comments.

**Comment by hmcl** [comment]
+1. Let's squash the commits and it's good to go as far as I am concerned. Thanks @srdo

File: dev-tools/release_notes.py (hunk 3 above)
Comment (ptgoetz) [inline dev-tools/release_notes.py]: Ah. Gotcha.

**Comment by ptgoetz** [comment]
+1

Minor nit: It generates html, but not an html *document* (i.e. missing head, body, etc. tags), but that's just a few print statements that can be added at merge time.

**Comment by srdo** [comment]
Added the missing html tags and a title.

**Comment by ptgoetz** [comment]
Still +1

**Comment by srdo** [comment]
Thanks, merging.

**Pull request merged**